### ISSUE TITLE
Can not performance a query with params

### ISSUE THREAD

**Issue by drouchy** [neo4jrb/neo4j-core#178]
Hi,

I can't perform a query with params with neo4j 2.1.7 (neo4j core 4.0.4) . When I copy the query from the doc, there is a bug:

```
[1] pry(#<Shutl::HealthCheckRoutes>)> n = Neo4j::Session.query("MATCH (n) WHERE ID(n) = #{0} RETURN n").first.n
=> EmbeddedNode neo_id: 0
[2] pry(#<Shutl::HealthCheckRoutes>)> n = Neo4j::Session.query("MATCH (n) WHERE ID(n) = {foobar} RETURN n", foobar: 0).first.n
NoMethodError: undefined method `merge' for "MATCH (n) WHERE ID(n) = {foobar} RETURN n":String
from /opt/boxen/rbenv/versions/jruby-1.7.12/lib/ruby/gems/shared/gems/neo4j-core-4.0.4/lib/neo4j-embedded/embedded_session.rb:116:in `query'
[3] pry(#<Shutl::HealthCheckRoutes>)>
```

Cheers,

David

**Comment by cheerfulstoic** [comment]
Interesting... Looking now

**Comment by cheerfulstoic** [comment]
I think I've figured it out.  Could you try changing/adding this to your `Gemfile`?

```ruby
  gem 'neo4j-core', github: 'neo4jrb/neo4j-core', branch: '178_fix'
```

If that works for you I'll cut a 4.0.5

**Comment by drouchy** [comment]
yep, that fixes the problem.

Thanks for the quick response


Cheers,

David

**Comment by cheerfulstoic** [comment]
Great!  I just released 4.0.5, so feel free to update to that instead of the github repo branch

**Comment by drouchy** [comment]
Hi,

any reason why this fix is not on master?

for some reasons, I need to bundle against the head of master, so my cypher queries fail due to this bug

Cheers.

David

**Comment by cheerfulstoic** [comment]
Sorry about losing track of this issue.  I've just made the fix from 4.0.5 on master and pushed
